packages/frontend/src/server/features/scaling/costs/utils/add-if-defined.test.ts

```diff
@@ -0,0 +1,76 @@
+import { expect } from 'earl'
+import { addIfDefined } from './add-if-defined'
+
+type TestCase = {
+  a: number | undefined | null
+  b: number | undefined | null
+  expected: number | undefined
+  description: string
+}
+
+const testCases: TestCase[] = [
+  {
+    a: 2,
+    b: 3,
+    expected: 5,
+    description: 'should return the sum of two defined numbers',
+  },
+  {
+    a: 2,
+    b: undefined,
+    expected: 2,
+    description: 'should return the first number if the second is undefined',
+  },
+  {
+    a: 2,
+    b: null,
+    expected: 2,
+    description: 'should return the first number if the second is null',
+  },
+  {
+    a: undefined,
+    b: 3,
+    expected: 3,
+    description: 'should return the second number if the first is undefined',
+  },
+  {
+    a: null,
+    b: 3,
+    expected: 3,
+    description: 'should return the second number if the first is null',
+  },
+  {
+    a: undefined,
+    b: undefined,
+    expected: undefined,
+    description: 'should return undefined if both numbers are undefined',
```

Comment:
Wouldn't zero be cleaner? To me this abstraction feels weird.

Reply:
It adds up only if something is defined. Take a look at the files that are using it - it would be nightmare to inline this, that's why I did this abstraction